shared/js/project.js

```diff
@@ -514,10 +514,11 @@ var projects = module.exports = {
    * @returns {jQuery.Deferred} A deferred which will resolve when the project loads.
    */
   load: function () {
-    var deferred;
+    var deferred = new $.Deferred();
```

Comment:
Changes to this method are _only_ trying to ensure that it always returns a `Deferred` that eventually resolves; before, there were several cases where it had no return value.